static/js/composebox_typeahead.js

```diff
@@ -788,26 +792,28 @@ export function show_flatpickr(element, callback, default_timestamp, options = {
         formatDate: (date) => formatISO(date),
         disableMobile: true,
         onKeyDown: (selectedDates, dateStr, instance, event) => {
-            const hotkey = get_keydown_hotkey(event);
-
-            if (["tab", "shift_tab"].includes(hotkey.name)) {
-                const elems = [
-                    instance.selectedDateElem,
-                    instance.hourElement,
-                    instance.minuteElement,
-                    instance.amPM,
-                    $(".flatpickr-confirm")[0],
-                ];
-                const i = elems.indexOf(event.target);
-                const n = elems.length;
-                const remain = (i + (event.shiftKey ? -1 : 1)) % n;
-                const target = elems[Math.floor(remain >= 0 ? remain : remain + n)];
-                event.preventDefault();
+            if (!is_numeric_key(event.key)) {
```

Comment:
I'd do an early return here, like you do in the other place.

Also, this doesn't fully fix the bug -- you can't use `backspace` when editing the number inputs.